patches/server/1047-Configurable-memory-condition-tick-rate.patch

```diff
@@ -0,0 +1,30 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: ruViolence <78062896+ruViolence@users.noreply.github.com>
+Date: Wed, 8 Nov 2023 01:25:45 +0800
+Subject: [PATCH] Configurable memory condition tick rate
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/ai/behavior/declarative/BehaviorBuilder.java b/src/main/java/net/minecraft/world/entity/ai/behavior/declarative/BehaviorBuilder.java
+index 6f323026b2d6fb5151e8b9dd9466e96eb026ccbd..6aeb432d7e79cbf5c64da934bd6d4d63a7a35e44 100644
+--- a/src/main/java/net/minecraft/world/entity/ai/behavior/declarative/BehaviorBuilder.java
++++ b/src/main/java/net/minecraft/world/entity/ai/behavior/declarative/BehaviorBuilder.java
+@@ -333,8 +333,19 @@ public class BehaviorBuilder<E extends LivingEntity, M> implements App<BehaviorB
+     static final class PureMemory<E extends LivingEntity, F extends K1, Value> extends BehaviorBuilder<E, MemoryAccessor<F, Value>> {
+         PureMemory(final MemoryCondition<F, Value> query) {
+             super(new BehaviorBuilder.TriggerWithResult<E, MemoryAccessor<F, Value>>() {
++                // Paper start - memory condition tick rate
++                private final String configKey = query.memory().toString().toLowerCase(java.util.Locale.ROOT);
```

Comment:
I think using `Locale.ENGLISH` is better here? We don't want it changing across systems which I *think* ROOT might?

Reply:
To quote the [Javadocs of the root locale](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Locale.html#ROOT):

> This is regarded as the base locale of all locales, and is used as the language/country neutral locale for the locale sensitive operations.

Reply:
I copied this from that commit. Are you sure I need to change it to `ENGLISH`?
https://github.com/PaperMC/Paper/commit/12942dc94ba42043461a4d5b5c049a6e53f7c2ee#diff-3e8d15a07d2e1a33d90f0ab38c54c0457d715330c7e478cb55e1ac794d8526a5R134

Reply:
Ok, sounds like I was under the wrong impression of what it was. `ROOT` is fine.